test/agent.js

```diff
@@ -0,0 +1,36 @@
+const tap = require('tap')
+const http = require('http')
+const { request } = require('../lib/agent')
+
+tap.test('request a resource using globalAgent', t => {
```

Comment:
Only enable for Node 14+

Reply:
was going to ask about this... what should I do in the actual implementation if they aren't using Node 14+? Load a polyfill dep?

Reply:
Just don't export it?

Reply:
This is going to be impossible to type. I propose we throw an error instead?

Reply:
Lol i'm copy/pasting the util i just wrote in case we need it later:
```js
function nodeMajorVersionIsGreaterThanOrEqualTo ({ major: expectedMajor, minor: expectedMinor, patch: expectedPatch }) {
  const [ foundMajor, foundMinor, foundPatch ] = process.versions.node.split('.').map(str => parseInt(str, 10))

  return foundMajor >= expectedMajor && foundMinor >= expectedMinor && foundPatch >= expectedPatch
}
```